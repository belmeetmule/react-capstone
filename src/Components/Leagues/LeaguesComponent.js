import React, { useEffect, useState } from 'react';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Spinner from 'react-bootstrap/Spinner';
import { getLeaguesData } from '../../Redux/League/LeaguesSlice';

const LeaguesComponent = () => {
  const [title, setTitle] = useState('');
  const dispatch = useDispatch();
  const { leagues } = useSelector((state) => state.leagues);

  useEffect(() => {
    dispatch(getLeaguesData());
  }, []);

  return (
    <>
      <div id="homePage_header" className="text-center">
        <h1 id="main-header-text">
          <span className="text-white">Soccer</span>
          {' '}
          Leagues
        </h1>
        <Form className="d-flex">
          <Form.Control
            type="search"
            placeholder="Search"
            className="me-2"
            aria-label="Search"
            onChange={(e) => setTitle(e.target.value)}
          />
          <Button variant="outline-success">Search</Button>
        </Form>
      </div>
      <div id="consumePage">
        {leagues ? (
          leagues
            .filter((leagues) => leagues.name.toLowerCase().match(title.toLocaleLowerCase()))
            .map((leagues) => (
              <Link key={leagues.id} to={`/detail/${leagues.id}`} className="m-list1">
                <div className="card">
                  <img className="card-img-top" src={leagues.logo} alt="logo" />
                  <div className="card-body">
                    <h5 className="card-title">{leagues.name}</h5>
                    <p className="card-text">{leagues.country}</p>
                  </div>
                </div>
              </Link>
            ))) : (
              <div style={{
                color: 'red', display: 'flex', faAlignJustify: 'center', fontSize: '40px',
              }}
              >
                <Spinner />
                no Data
              </div>
        )}
      </div>
    </>
  );
};
export default LeaguesComponent;
