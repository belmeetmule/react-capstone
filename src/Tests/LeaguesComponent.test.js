import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import LeaguesComponent from '../components/Leagues/LeaguesComponent';
import store from '../Redux/configureStore';

describe('Testing Leagues component', () => {
  it('Should render Leagues', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <LeaguesComponent />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
