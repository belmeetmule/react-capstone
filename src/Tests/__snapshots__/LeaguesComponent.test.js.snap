// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Leagues component Should render Leagues 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center"
        id="homePage_header"
      >
        <h1
          id="main-header-text"
        >
          <span
            class="text-white"
          >
            Soccer
          </span>
           
          Leagues
        </h1>
        <form
          class="d-flex"
        >
          <input
            aria-label="Search"
            class="me-2 form-control"
            placeholder="Search"
            type="search"
          />
          <button
            class="btn btn-outline-success"
            type="button"
          >
            Search
          </button>
        </form>
      </div>
      <div
        id="consumePage"
      >
        <div
          style="color: red; display: flex; font-size: 40px;"
        >
          <div
            class="spinner-undefined"
          />
          no Data
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center"
      id="homePage_header"
    >
      <h1
        id="main-header-text"
      >
        <span
          class="text-white"
        >
          Soccer
        </span>
         
        Leagues
      </h1>
      <form
        class="d-flex"
      >
        <input
          aria-label="Search"
          class="me-2 form-control"
          placeholder="Search"
          type="search"
        />
        <button
          class="btn btn-outline-success"
          type="button"
        >
          Search
        </button>
      </form>
    </div>
    <div
      id="consumePage"
    >
      <div
        style="color: red; display: flex; font-size: 40px;"
      >
        <div
          class="spinner-undefined"
        />
        no Data
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
